langchain4j-mariadb/src/main/java/dev/langchain4j/store/embedding/mariadb/ColumnsMetadataHandler.java

```diff
@@ -0,0 +1,152 @@
+
+import dev.langchain4j.data.document.Metadata;
+import dev.langchain4j.store.embedding.filter.Filter;
+import java.sql.*;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * Handle Metadata stored in independent columns
+ */
+class ColumnsMetadataHandler implements MetadataHandler {
+
+    final List<MetadataColumDefinition> columnsDefinition;
+    final List<String> columnsName;
+    final List<String> escapedColumnsName;
+
+    final MariaDbFilterMapper filterMapper;
+    final List<String> indexes;
+
+    /**
+     * MetadataHandler constructor
+     * @param config {@link MetadataStorageConfig} configuration
+     */
+    public ColumnsMetadataHandler(MetadataStorageConfig config, List<String> sqlKeywords) {
+        List<String> columnsDefinitionList =
+                ensureNotEmpty(config.columnDefinitions(), "Metadata definition");
+        this.filterMapper = new ColumnFilterMapper();
+
+        // enquote identifier if keywords
+        this.indexes =
+                ((List<String>) getOrDefault(config.indexes(), Collections.emptyList()))
+                        .stream()
+                                .map(
+                                        key -> {
+                                            if (sqlKeywords.contains(
+                                                    key.toLowerCase(Locale.ROOT))) {
+                                                try {
+                                                    return org.mariadb.jdbc.Driver
+                                                            .enquoteIdentifier(key, true);
+                                                } catch (SQLException e) {
+                                                    // eat
+                                                }
+                                            }
+                                            return key;
+                                        })
+                                .toList();
+        this.columnsDefinition =
+                columnsDefinitionList.stream()
+                        .map(str -> MetadataColumDefinition.from(str, sqlKeywords))
+                        .collect(Collectors.toList());
+        this.escapedColumnsName =
+                columnsDefinition.stream()
+                        .map(MetadataColumDefinition::getEscapedName)
+                        .collect(Collectors.toList());
+        this.columnsName =
+                columnsDefinition.stream()
+                        .map(MetadataColumDefinition::getName)
+                        .collect(Collectors.toList());
+    }
+
+    @Override
+    public String columnDefinitionsString() {
+        return this.columnsDefinition.stream()
+                .map(MetadataColumDefinition::getFullDefinition)
+                .collect(Collectors.joining(","));
```

Comment:
nit: this can be cached

Reply:
have not done it, since it's used only one time, when creating table